patches/server/0860-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,459 @@
+ import org.bukkit.craftbukkit.inventory.CraftEntityEquipment;
+ import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.craftbukkit.potion.CraftPotionUtil;
+@@ -72,6 +75,7 @@ import org.bukkit.entity.TippedArrow;
+ import org.bukkit.entity.Trident;
+ import org.bukkit.entity.WitherSkull;
+ import org.bukkit.entity.memory.MemoryKey;
++import org.bukkit.event.entity.EntityDamageEvent;
+ import org.bukkit.event.entity.EntityPotionEffectEvent;
+ import org.bukkit.event.player.PlayerTeleportEvent;
+ import org.bukkit.inventory.EntityEquipment;
+@@ -345,7 +349,7 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+     // Paper End - Bee Stinger API
+     @Override
+     public void damage(double amount) {
+-        this.damage(amount, null);
++        this.damage(amount, (org.bukkit.entity.Entity) null); // Paper - the typecast
+     }
+ 
+     @Override
+@@ -363,6 +367,19 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         entity.hurt(reason, (float) amount);
+     }
+ 
++    // Paper start
++    @Override
++    public void damage(double amount, EntityDamageEvent.DamageCause cause, Entity source) {
++        net.minecraft.world.entity.Entity handle = null;
++        if (source != null) {
++            handle = ((CraftEntity) source).getHandle();
++        }
++        DamageSource reason = DamageUtil.convertDamageCauseToSource(cause, handle);
++        CraftEventFactory.entityDamage = handle;
++        entity.hurt(reason, (float) amount);
++    }
++    // Paper end
++
+     @Override
+     public Location getEyeLocation() {
+         Location loc = getLocation();
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 519b17fac445b7118f5493508bddccd368dadcde..f3758b8e3ac6ed609b71bd53708a01dc15f0f534 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -2,7 +2,6 @@ package org.bukkit.craftbukkit.event;
+ 
+ import com.google.common.base.Function;
+ import com.google.common.base.Functions;
+-import com.google.common.collect.Lists;
+ import com.mojang.datafixers.util.Either;
+ import java.net.InetAddress;
+ import java.util.ArrayList;
+@@ -12,6 +11,8 @@ import java.util.List;
+ import java.util.Map;
+ import java.util.stream.Collectors;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.util.DamageUtil;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.Direction;
+ import net.minecraft.network.protocol.game.ServerboundContainerClosePacket;
+@@ -76,7 +77,6 @@ import org.bukkit.craftbukkit.block.CraftBlock;
+ import org.bukkit.craftbukkit.block.CraftBlockState;
+ import org.bukkit.craftbukkit.block.CraftBlockStates;
+ import org.bukkit.craftbukkit.block.data.CraftBlockData;
+-import org.bukkit.craftbukkit.entity.CraftEntity;
```

Comment:
fully qualified names + remove random whitespace + don't need to delete imports I think